pubsub/google/cloud/pubsub_v1/publisher/_batch/thread.py

```diff
@@ -187,56 +187,62 @@ def _commit(self):
                 _LOGGER.debug("Batch is already in progress, exiting commit")
                 return
 
-            # Sanity check: If there are no messages, no-op.
-            if not self._messages:
-                _LOGGER.debug("No messages to publish, exiting commit")
-                self._status = base.BatchStatus.SUCCESS
-                return
-
-            # Begin the request to publish these messages.
-            # Log how long the underlying request takes.
-            start = time.time()
-
-            try:
-                response = self._client.api.publish(self._topic, self._messages)
-            except google.api_core.exceptions.GoogleAPIError as exc:
-                # We failed to publish, set the exception on all futures and
-                # exit.
-                self._status = base.BatchStatus.ERROR
-
-                for future in self._futures:
-                    future.set_exception(exc)
+        # Once in the IN_PROGRESS state, no other thread can publish additional
+        # messages or initiate a commit (those operations becme a no-op), thus
+        # it is safe to release the state lock here. Releasing the lock avoids
+        # blocking other threads in case api.publish() below takes a long time
+        # to complete.
```

Comment:
Link to the GitHub issue too, for more context?

Reply:
It may be good to have a similar comment on operations that add messages / make a commit so that they continue to be a no-op when in `IN_PROGRESS` state.

Reply:
Could be useful to some, will add that.